amundsen_application/api/search/v0.py

```diff
@@ -109,11 +122,8 @@ def _search(*, search_term: str, page_index: int) -> Dict[str, Any]:
 
     TODO: Define an interface for envoy_client
     """
-    results_dict = {
-        'results': [],
+    results_dict: Dict[str, Any] = {
```

Comment:
The moment I nested the objects, Mypy started complaining. I had tried initializing the way you described but there were some errors when I tried to assign the results. Maybe it was because I was trying to modify results after assignment. I will try again.